File: 3379-transformed-array/3379-transformed-array.java
class Solution {
    public int[] constructTransformedArray(int[] nums) {
        int n = nums.length;
        int[] ans = new int[n];

        for (int i = 0; i < n; i++) {
            int newIndex = ((i + nums[i]) % n + n) % n;
            ans[i] = nums[newIndex];
        }

        return ans;
    }
}

File: 3379-transformed-array/3379-transformed-array.py
class Solution:
    def constructTransformedArray(self, nums: List[int]) -> List[int]:
        n = len(nums)
        ans = [0] * n

        for i in range(n):
            index = (i + nums[i]) % n
            ans[i] = nums[index]

        return ans
